test/jdk/java/foreign/valist/VaListTest.java

```diff
@@ -674,6 +693,17 @@ public void testCopyUnusableAfterOriginalClosed(Function<Consumer<VaList.Builder
                     assertEquals((long) VH_BigPoint_x.get(struct), 8);
                     assertEquals((long) VH_BigPoint_y.get(struct), 16);
                 })},
+                { linkVaListCB("upcallBigStructPlusScalar"), VaListConsumer.mh(vaList -> {
+                    MemorySegment struct = vaList.vargAsSegment(BigPoint_LAYOUT, ResourceScope.newImplicitScope());
+                    assertEquals((long) VH_BigPoint_x.get(struct), 8);
+                    assertEquals((long) VH_BigPoint_y.get(struct), 16);
+
+                    assertEquals((long) vaList.vargAsLong(C_LONG), 42);
+                })},
+                { linkVaListCB("upcallBigStructPlusScalar"), VaListConsumer.mh(vaList -> {
+                    vaList.skip(BigPoint_LAYOUT);
+                    assertEquals((long) vaList.vargAsLong(C_LONG), 42);
+                })},
```

Comment:
Looks like a carrier size mismatch here:

```
java.lang.IllegalArgumentException: Carrier size mismatch: long != b32[abi/kind=LONG]
        at jdk.incubator.foreign/jdk.internal.foreign.Utils.checkPrimitiveCarrierCompat(Utils.java:111)
        at jdk.incubator.foreign/jdk.internal.foreign.abi.SharedUtils.checkCompatibleType(SharedUtils.java:231)
        at jdk.incubator.foreign/jdk.internal.foreign.abi.x64.windows.WinVaList.read(WinVaList.java:114)
        at jdk.incubator.foreign/jdk.internal.foreign.abi.x64.windows.WinVaList.read(WinVaList.java:109)
        at jdk.incubator.foreign/jdk.internal.foreign.abi.x64.windows.WinVaList.vargAsLong(WinVaList.java:84)
        at VaListTest.lambda$upcalls$58(VaListTest.java:705)
        at jdk.incubator.foreign/jdk.internal.foreign.abi.ProgrammableInvoker.invokeNative(Native Method)
        at jdk.incubator.foreign/jdk.internal.foreign.abi.ProgrammableInvoker.invokeMoves(ProgrammableInvoker.java:290)
        at VaListTest.testUpcall(VaListTest.java:530)
```

Need to use `C_LONG_LONG` to be cross-platform compatible. (sorry for not noticing this).

```suggestion
                { linkVaListCB("upcallBigStructPlusScalar"), VaListConsumer.mh(vaList -> {
                    MemorySegment struct = vaList.vargAsSegment(BigPoint_LAYOUT, ResourceScope.newImplicitScope());
                    assertEquals((long) VH_BigPoint_x.get(struct), 8);
                    assertEquals((long) VH_BigPoint_y.get(struct), 16);

                    assertEquals((long) vaList.vargAsLong(C_LONG_LONG), 42);
                })},
                { linkVaListCB("upcallBigStructPlusScalar"), VaListConsumer.mh(vaList -> {
                    vaList.skip(BigPoint_LAYOUT);
                    assertEquals((long) vaList.vargAsLong(C_LONG_LONG), 42);
                })},
```

Reply:
Also, it looks like the cast to `(long)` on those lines is redundant (thanks IntelliJ).

Reply:
Thanks for your help! These are both fixed now.